src/imgproc/bee_eye.cc

```diff
@@ -0,0 +1,48 @@
+// BoB robotics includes
+#include "common/bob_robotics_path.h"
+#include "common/macros.h"
+#include "imgproc/bee_eye.h"
+
+// Standard C includes
+#include <cmath>
+
+namespace BoBRobotics {
+namespace ImgProc {
+namespace BeeEye {
+
+Map::Map(const float eyeData[][4], const int eyeDataLength,
```

Comment:
I think this copies the arrays by value which, especially with the eye data is probably not what you want. The syntax here is singularly grim and, I find, very hard to remember but I **think** this should be the correct syntax:
```c++
template <size_t eyeDataLength>
Map::Map(const float (&eyeData)[eyeDataLength][4], const int (&eyeSize)[2], const int (&imageSize)[2], bool useMask)
```
On the bright side, it removes the need for manually handling eyeDataLength but, on the less-bright side it would need to go in the header.

Reply:
Oh yeah, of course. I forgot that fixed-size arrays get passed by value.

Reply:
Hmm I tried that and then was getting an error saying I was trying to declare an array of references... So I just changed it to ``float (*eyeData)[4]`` instead and that seemed to work.

Reply:
How very strange - http://cpp.sh/96pyq works fine for me

Reply:
Ok, so I've got it to work now.

I'm not sure what I was doing wrong. It might have been that there was an earlier error that I missed that was causing this. It's still a bit strange.

Reply:
Awesome! As I said, that is one of my least-favourite bits of C++  syntax, so weird-lookin'